### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by hendrik-skubch**
- First iteration on runningPlan
- update to clang format

### CODE REVIEW

File: alica_engine/src/engine/RunningPlan.cpp [hunk 1]
```diff
@@ -36,86 +36,98 @@ using std::shared_ptr;
...
-    this->robotsAvail = unique_ptr<list<const supplementary::AgentID*>>(new list<const supplementary::AgentID*>);
 }
 
+RunningPlan::RunningPlan(AlicaEngine* ae)
+    : _ae(ae)
+    , _planType(nullptr)
+    , _plan(nullptr)
+    , _activeState(nullptr)
+    , _activeEntryPoint(nullptr)
+    , _behaviour(false)
```
Comment (gauthamm) [inline alica_engine/src/engine/RunningPlan.cpp]: I suppose this variable is unused now

File: alica_engine/src/engine/RunningPlan.cpp [hunk 2]
```diff
@@ -36,86 +36,98 @@ using std::shared_ptr;
...
-    this->plan = bc;
-    this->bp = ae->getBehaviourPool();
-    this->behaviour = true;
-}
+    : _ae(ae)
+    , _planType(nullptr)
+    , _plan(bc)
+    , _activeState(nullptr)
+    , _activeEntryPoint(nullptr)
+    , _behaviour(true)
```
Comment (gauthamm) [inline alica_engine/src/engine/RunningPlan.cpp]: unused variable

File: alica_engine/include/engine/parser/tinyxml2.h [hunk 3]
```diff
@@ -1250,7 +1262,7 @@ class TINYXML2_LIB XMLElement : public XMLNode {
         doesn't exist. It is overloaded for the primitive types,
                 and is a generally more convenient replacement of
                 QueryIntAttribute() and related functions.
-                
+
```
Comment (gauthamm) [inline alica_engine/include/engine/parser/tinyxml2.h]: what's this?

File: alica_engine/src/engine/RunningPlan.cpp (hunk 1 above)
Comment (hendrik-skubch) [reply alica_engine/src/engine/RunningPlan.cpp]: No, it is very much used.

File: alica_engine/include/engine/parser/tinyxml2.h (hunk 3 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/parser/tinyxml2.h]: formatting?

**Pull request merged**